src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/ParameterNamesCannotChangeTests.cs

```diff
@@ -0,0 +1,96 @@
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Microsoft.DotNet.ApiCompatibility.Tests;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules.Tests
+{
+    public class ParameterNamesCannotChangeTests
+    {
+        public static TheoryData<string, string, CompatDifference[]> TestCases => new()
+        {
+            // Method
+            {
+                 @"
+namespace CompatTests
+{
+  public class First {
+    public void F(int a, string s) {}
+  }
+}
+",
+                 @"
+namespace CompatTests
+{
+  public class First {
+    public void F(int b, string t) {}
+  }
+}
+",
+                 new CompatDifference[] {
+                     CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotChangeParameterName, "", DifferenceType.Changed, "M:CompatTests.First.F(System.Int32,System.String)$0"),
+                     CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotChangeParameterName, "", DifferenceType.Changed, "M:CompatTests.First.F(System.Int32,System.String)$1")
+                 }
+            },
+            // Constructor
+            {
+                 @"
+namespace CompatTests
+{
+  public class First {
+    public First(int a, string s) {}
+  }
+}
+",
+                 @"
+namespace CompatTests
+{
+  public class First {
+    public First(int b, string t) {}
+  }
+}
+",
+                 new CompatDifference[] {
+                     CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotChangeParameterName, "", DifferenceType.Changed, "M:CompatTests.First.#ctor(System.Int32,System.String)$0"),
+                     CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotChangeParameterName, "", DifferenceType.Changed, "M:CompatTests.First.#ctor(System.Int32,System.String)$1")
+                 }
+            },
+            // Property
+            {
+                 @"
+namespace CompatTests
+{
+  public class First {
+    public int F { get; }
+  }
+}
+",
+                 @"
+namespace CompatTests
+{
+  public class First {
+    public int F { get; }
+  }
+}
+",
+                 new CompatDifference[] {}
+            }
+        };
+
+        [Theory]
+        [MemberData(nameof(TestCases))]
+        public void EnsureDiagnosticIsReported(string leftSyntax, string rightSyntax, CompatDifference[] expected)
+        {
+            TestRuleFactory s_ruleFactory = new((settings, context) => new CannotChangeParameterName(settings, context, true));
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new(s_ruleFactory);
+            IEnumerable<CompatDifference> actual = differ.GetDifferences(left, right);
+            Assert.Equal(expected, actual);
```

Comment:
For readability please follow the "Array Act Assert" best practice and separate the logical from each other:

```suggestion
            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
            TestRuleFactory s_ruleFactory = new((settings, context) => new CannotChangeParameterName(settings, context, true));
            ApiComparer differ = new(s_ruleFactory);

            IEnumerable<CompatDifference> actual = differ.GetDifferences(left, right);

            Assert.Equal(expected, actual);
```